src/Examine.AzureDirectory/AzureDirectory.cs

```diff
@@ -226,31 +276,55 @@ public override long FileLength(string name)
                 return CacheDirectory.FileLength(name);
             }
 
-            if (TryGetBlobFile(name, out var blob, out var err))
+            try
             {
-                Debugger.Launch();
+                var blob = _blobContainer.GetBlockBlobReference(RootFolder + name);
+                blob.FetchAttributes();
 
-                // index files may be compressed so the actual length is stored in metatdata
+                // index files may be compressed so the actual length is stored in metadata
                 var hasMetadataValue = blob.Metadata.TryGetValue("CachedLength", out var blobLegthMetadata);
 
                 if (hasMetadataValue && long.TryParse(blobLegthMetadata, out var blobLength))
                 {
                     return blobLength;
                 }
-                return blob.Properties.Length; // fall back to actual blob size
+
+                // fall back to actual blob size
+                return blob.Properties.Length;
             }
-            else
+            catch (Exception e)
             {
-                // NOTE: We cannot return 0 or something like that, this will end up throwing an exception in Lucene:
-                // https://github.com/apache/lucenenet/blob/3.0.3/src/core/Index/TermVectorsTermsWriter.cs#L115
+                //  Sync(name);
+                return CacheDirectory.FileLength(name);
+            }
+        }
 
-                // This method will be called after committing the index. Since the local index can be directly written to
-                // we need to return the real result.
+        public override void Sync(string name)
```

Comment:
Can we have notes on what this is doing and when Sync is called?

Reply:
Sure will add